x/auth/signing/verify.go

```diff
@@ -57,6 +59,25 @@ func internalSignModeToAPI(mode signing.SignMode) (signingv1beta1.SignMode, erro
 	}
 }
 
+// VerifysignatureCache is a cache of verified signatures
+// If the cached item is not found, it will be verified and added to the cache
+func verifySig(signBytes, sig []byte, pubKey cryptotypes.PubKey) bool {
+	bz := sha256.Sum256(append(signBytes, sig...))
+	hash := bz[:]
+	cachePub, ok := SignatureCache().Get(hash)
+	if ok {
+		SignatureCache().Remove(hash)
```

Comment:
Can we document that this caching performs one time caching then purging? Was this intended to remove the keyed value as soon as it is found? The PR description and docs don't state this.

Reply:
Seems like it should only purge on success as well

Reply:
on succesful verifiaction or?